pwr.md

```diff
@@ -0,0 +1,147 @@
+* `maximum`:            Set all cores minimum to 3900Mhz and maximum to 3900Mhz.
+* `base`:               Set all cores minimum to 2300Mhz and maximum to 2300Mhz.
+* `default`:            Set all cores minimum to 800Mhz and maximum to 3900Mhz.
+* `no_turbo`:           Set all cores minimum to 800Mhz and maximum to 2300Mhz.
+* `sst_bf_base`:        Set high priority cores minimum and maximum to 2700Mhz, normal priority cores minimum and maximum to 2100Mhz (Only available with SST-BF).
+* `sst_bf_high_turbo`:  Set high priority cores minimum to 2700Mhz and maximum to 3900Mhz, normal priority cores minimum to 800Mhz and maximum to 2100Mhz(Only available with SST-BF).
+* `sst_bf_low_turbo`:   Set high priority cores minimum to 2700Mhz and maximum to 2700Mhz, normal priority cores minimum to 800Mhz and maximum to 3900Mhz(Only available with SST-BF).
+>Specific frequencies will depend on system and configuration.
+
+```python
+For c in cores:
+    c.commit("sst_bf_base") # Configure cores with the SST-BF configuration
+```
+
+## Refreshing cpu stats
+CPU stats can become out of date, such as `curr_freq` or `sst_bf_configured`. These can be refreshed in with `refresh_stats()`.
+It is *advised* that CPU and core objects be updated with `refresh_stats()` before requesting object data.
+
+core.refresh_stats() will update:
+* `curr_freq`             
+* `min_freq`                
+* `max_freq`                
+* `epp`
+                
+cpu.refresh_stats() will update:
+* `uncore_freq`
+* `uncore_max_freq` 
+* `uncore_min_freq`
+* `sst_bf_configured`
+
+```python
+for c in cores:
+    c.refresh_stats()
+    if not c.sst_bf_enabled:
+        c.commit("sst_bf_base")
+```
+
+## Object Referencing
+Both CPU & core objects reference each other, the list of core objects can be accessed through the CPU and the CPU object of a core can be accessed through a core.
+
+```python
+for core in cores:  # Loop through cores in object list 
+    if core.cpu.sst_bf_enabled:  # Access CPU flag from core object
+        if core.high_priority:  # Check if core has high priority base frequency 
+            core.min_freq = core.base_freq  # Modify minimum desired frequency
+            core.max_freq = core.sst_bf_base_freq  # Modify maximum desired frequency
+            core.commit()  # Commit changes to hardware
+```
+or
+```python
+for cpu in cpus:
+    if cpu.core_list[0].epp == "performance":  # Access core performance profile through core list of CPU object
+        cpu.uncore_max_freq = cpu.uncore_hw_max  # set desired max uncore frequency from hardware limit
+        cpu.uncore_min_freq = cpu.uncore_hw_min  # set desired min uncore frequency from hardware limit
+        cpu.commit() #Commit changes to hardware
+```
+
+## Power Consumption
+The power consumption of the CPU can be read from the `power_consumption` attribute, this value can be compared against the `tdp` value to check is the current power draw close to the limit, indicated by the tdp value.
```

Comment:
i would add the following:
```
The power consumption is reported as average since last time it was calculated. The first time this value is read, it will be 0. The next time it is read, it will show average power consumption (in Watts) since last read. Time period between reads must not exceed 60 seconds, otherwise the value will be reset to 0.
```

Reply:
Does this mean the user will have to read it twice the first time? or is that initial read done on initialization? 
or is it more, the use case of this is to set a period of which you would like to measure power consumption i.e set a beginning and end point, rather than just getting the consumption at that very point.

this is turning into a rant, but I assume we cannot get a value for power consumption at a single instance, just a relative one?

Reply:
the initial read _is_ done on initialization, but in the general case, yes, if you want to measure power consumption over a period of time, you need to measure twice - that's kind of the definition of an "average" :)

there is no way to read "current power consumption" from the CPU - it is only possible to read how much power it has consumed thus far.

Reply:
Got it Thanks!

Reply:
actually, i'm now not so sure it is done on init. Do we call refresh_stats() on init, or do we initialize values internally, without specifically calling the function?

Reply:
no we call refresh_stats, which is better i guess.

Reply:
then it's done on init. (and it's better!)

Reply:
do we still want to say "The first time this value is read, it will be 0.", since we dont know this for sure,

Reply:
_can_ be 0? it would depend on when it was read. i.e. if you initialize the library but only do the first refresh a minute later?

Reply:
exactly, okay I'll word smith it. :)